modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/Processing.scala

```diff
@@ -237,21 +235,18 @@ object Processing {
     realTimeWindowStarted: FiniteDuration
   ): Stream[F, Unique.Token] =
     Stream.eval(ref.get).flatMap { state =>
-      val commit = NonEmptyList.fromList(state.framesOnDisk) match {
-        case Some(nel) =>
-          val eventCount = state.framesOnDisk.map(_.count).sum
-          for {
-            _ <- Logger[F].info(s"Ready to Write and commit $eventCount events to the lake.")
-            _ <- Logger[F].info(s"Non atomic columns: [${state.nonAtomicColumnNames.toSeq.sorted.mkString(",")}]")
-            _ <- env.cpuPermit.surround(env.lakeWriter.commit(nel))
-            now <- Sync[F].realTime
-            _ <- Logger[F].info(s"Finished writing and committing $eventCount events to the lake.")
-            _ <- env.metrics.addCommitted(eventCount)
-            _ <- env.metrics.setProcessingLatency(now - realTimeWindowStarted)
-          } yield ()
-        case None =>
-          Logger[F].info("A window yielded zero good events.  Nothing will be written into the lake.")
-      }
+      val commit = if (state.numEvents > 0) {
+        for {
+          _ <- Logger[F].info(s"Window ${state.viewName} ready to write and commit ${state.numEvents} events to the lake.")
+          _ <- Logger[F].info(s"Non atomic columns: [${state.nonAtomicColumnNames.toSeq.sorted.mkString(",")}]")
+          _ <- env.lakeWriter.commit(state.viewName)
+          now <- Sync[F].realTime
+          _ <- Logger[F].info(s"Window ${state.viewName} finished writing and committing ${state.numEvents} events to the lake.")
+          _ <- env.metrics.addCommitted(state.numEvents)
+          _ <- env.metrics.setProcessingLatency(now - realTimeWindowStarted)
```

Comment:
I know this line hasn't changed, but is it OK that we set processing latency only when there are some good events? Maybe that should be separate `Pipe` in `processBatches` where we set metrics regardless of event counts?

Reply:
Processing latency is interesting because it is only defined after writing some events.  For example, imagine we have a windowing of 5 minutes, so we only commit events every 5 minutes.  And imagine we emit metrics every 1 minute.  The processing metric over time looks like this:

```
12:01:00 undefined
12:02:00 undefined
12:03:00 undefined
12:04:00 undefined
12:05:00 320 seconds
12:06:00 undefined
12:07:00 undefined
12:08:00 undefined
12:09:00 undefined
12:10:00 340 seconds
```

So... sure we could emit a zero instead of undefined.  But it doesn't really mean anything, and it is even misleading.  E.g. at 12:07:00 is the processing latency really zero?

This metric mainly exists to help us developers understand how the lake loader is behaving.  So we are free to make the metric behave however we find helpful.

Reply:
Yes, emitting zeros is not useful. But I thought more about bad data. Let's say for some reason loader produces only badrows, no good data. Then we don't have this processing latency set, right? So loader has done some processing, but its latency is not reported. Just a thought, not sure that would be really useful 🤷

Reply:
Oh I see what you mean now.

Hmmm... probably the same argument still applies.  If there is no good data for an entire window, then "processing latency" is not well defined.  I cannot think how I would benefit from seeing processing latency if all data went to bad.

If we cared about latency of data going to bad, then it should be a different metric and we should emit it every time there is bad data.  Currently, though, I don't care about the latency of data going to bad.